lib/galaxy/files/sources/__init__.py

```diff
@@ -71,6 +71,8 @@ def get_writable(self):
         return self.writable
 
     def user_has_access(self, user_context) -> bool:
+        if user_context is None and self.user_context_required:
+            return False
```

Comment:
But that means tool exports to restricted locations won't work, even if the user is allowed to access the file source.
So either we also serialize the user context, or we build a variant that skips the validation, or when we serialize the file sources for tools we remove the restriction (or maybe some other thing ?).
Removing the restriction could be done here after checking that the user has access: https://github.com/mvdbeek/galaxy/blob/dev/lib/galaxy/files/sources/__init__.py#L118

Reply:
Actually, this currently should work, because the method that is really called when trying to operate the file source is:
https://github.com/galaxyproject/galaxy/pull/12607/files#diff-644fe18408381b506c4abc27ffe07d42b8a8dc55a5a833f705662f65fe60c6d6L168
Which only checks the user access if the `user_context` is present.

I think serializing the user context in all scenarios will be the safest option, but I don't know how to do it yet :/